tests/foreman/installer/test_installer.py

```diff
@@ -1321,6 +1324,193 @@ def extract_help(filter='params'):
+    """Install Satellite with default options"""
+    installer_args = [
+        'scenario satellite',
+        f'foreman-initial-admin-password {settings.server.admin_password}',
+    ]
+    install_satellite(module_sat_ready_rhel[0], installer_args)
+    yield module_sat_ready_rhel[0]
+
+
+@pytest.fixture(scope='module')
+def sat_non_default_install(module_sat_ready_rhel):
+    """Install Satellite with various options"""
+    installer_args = [
+        'scenario satellite',
+        f'foreman-initial-admin-password {settings.server.admin_password}',
+        'foreman-rails-cache-store type:redis',
+        'foreman-proxy-content-pulpcore-hide-guarded-distributions false',
+    ]
+    install_satellite(module_sat_ready_rhel[1], installer_args)
+    yield module_sat_ready_rhel[1]
+
+
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.parametrize(
+    "module_sat_ready_rhel", [settings.server.version.rhel_version], indirect=True
+)
+def test_capsule_installation(sat_default_install, cap_ready_rhel, default_org):
+    """Run a basic Capsule installation
+
+    :id: 64fa85b6-96e6-4fea-bea4-a30539d59e65
+
+    :steps:
+        1. Get a Satellite
+        2. Configure capsule repos
+        3. Enable capsule module
+        4. Install and setup capsule
+
+    :expectedresults:
+        1. Capsule is installed and setup correctly
+
+    :CaseImportance: High
+    """
+    # Setup Capsule
+    org = sat_default_install.api.Organization().search(query={'search': f'name="{DEFAULT_ORG}"'})[
+        0
+    ]
+    setup_capsule(sat_default_install, cap_ready_rhel, org)
+    assert sat_default_install.api.Capsule().search(
+        query={'search': f'name={cap_ready_rhel.hostname}'}
+    )[0]
+    result = cap_ready_rhel.execute(
+        r'grep "\[ERROR" --after-context=100 /var/log/foreman-installer/satellite.log'
+    )
+    assert len(result.stdout) == 0
+    result = cap_ready_rhel.execute(
+        r'grep "\[ERROR" --after-context=100 /var/log/foreman-installer/capsule.log'
+    )
+    assert len(result.stdout) == 0
+    result = cap_ready_rhel.cli.Health.check()
+    assert 'FAIL' not in result.stdout
+
+
+@pytest.mark.e2e
+@pytest.mark.tier1
+@pytest.mark.parametrize(
+    "module_sat_ready_rhel", [settings.server.version.rhel_version], indirect=True
+)
+def test_foreman_rails_cache_store(sat_non_default_install):
+    """Test foreman-rails-cache-store option
+
+    :id: 379a2fe8-1085-4a7f-8ac3-24c421412f12
+
+    :steps:
+        1. Install Satellite.
+        2. Verify that foreman-redis package is installed.
+        3. Check /etc/foreman/settings.yaml
+
+    :CaseImportance: Medium
+
+    :customerscenario: true
+
+    :BZ: 2063717, 2165092
+    """
+    # Verify foreman-rails-cache-store option works
+    assert sat_non_default_install.execute('rpm -q foreman-redis').status == 0
+    assert (
+        sat_non_default_install.execute('grep "type: redis" /etc/foreman/settings.yaml').status == 0
+    )
```

Comment:
Could you perhaps implement a helper function that would read remote YAML file and return its contents?
This way you could validate the exact path in the YAML file.

Also can `/etc/foreman/settings.yaml` be a constant?

```python
def load_remote_yaml_file(self, file_path):
    """Load a remote yaml file and return a Box object"""
    data = self.session.sftp_read(file_path, return_data=True)
    return Box(yaml.load(data, yaml.FullLoader))
```

```suggestion
   settings_file = sat_non_default_install.load_remote_yaml_file('/etc/foreman/settings.yaml')
   assert settings_file.the.desired.path.where.the.value.is == 'redis'
```